tools/topology/topology2/cavs-nocodec.conf

```diff
@@ -753,3 +668,113 @@ Object.Base.route [
 		sink	"smart_amp.2.1"
 	}
 ]
+
+# There is pinmux conflict between SSP1 and DMIC on MTL,
+# so include SSP1 pipelines conditionally.
+IncludeByKey.SSP1_ENABLED {
+	"true" {
+		Object.Dai.SSP [
+			{
+				id 		1
+				dai_index	1
+				direction	"duplex"
+				name		NoCodec-1
+				default_hw_conf_id	0
+				sample_bits		32
+				quirks			"lbm_mode"
+				io_clk		$MCLK
+
+				Object.Base.hw_config.1 {
+					name	"SSP1"
+					id	0
+					mclk_freq	$MCLK
+					bclk_freq	3072000
+					tdm_slot_width	32
+				}
+			}
+		]
+
+		Object.Pipeline.host-copier-gain-mixin-playback [
+			{
+				index	3
+
+				Object.Widget.copier.1 {
+					stream_name 'SSP1 Playback'
+				}
+				Object.Widget.gain.1 {
+					Object.Control.mixer.1 {
+						name 'Playback Volume 3'
+					}
+				}
+			}
+		]
+
+		Object.Pipeline.mixout-gain-dai-copier-playback [
+			{
+				index	4
+
+				Object.Widget.copier.1 {
+					dai_index 1
+					dai_type "SSP"
+					copier_type "SSP"
+					stream_name "NoCodec-1"
+					node_type $I2S_LINK_OUTPUT_CLASS
+				}
+
+				Object.Widget.gain.1 {
+					Object.Control.mixer.1 {
+						name 'Main Playback Volume 4'
+					}
+				}
+			}
+		]
+
+		Object.Pipeline.passthrough-capture [
+			{
+				index 	9
+
+				Object.Widget.copier.1 {
+					stream_name 'SSP1 Capture'
+				}
+			}
+		]
+
+		Object.PCM.pcm [
+			{
+				name	"Port1"
+				id 1
+				direction	"duplex"
+				Object.Base.fe_dai.1 {
+					name	"Port1"
+				}
+
+				Object.PCM.pcm_caps.1 {
+					direction	"playback"
+					name "SSP1 Playback"
+					formats 'S16_LE,S24_LE,S32_LE'
+				}
+
+				Object.PCM.pcm_caps.2 {
+					direction	"capture"
+					name "SSP1 Capture"
+					formats 'S16_LE,S24_LE,S32_LE'
+				}
+			}
+		]
+
+		Object.Base.route [
+			{
+				source	"mixin.3.1"
+				sink	"mixout.4.1"
+			}
+			{
+				source	"gain.4.1"
+				sink	"copier.SSP.4.1"
+			}
+			{
+				source	"copier.SSP.10.1"
+				sink	"copier.host.9.1"
```

Comment:
this is far from self-explanatory, @ranj063's cleanups should really be applied, There's nothing in there that looks like SSP1....

Reply:
agree, The pipelines are group by pipeline instead by DAI, a clean up would be better.